FilesetRelPathWrapperTest.test_glob_mid_single is flaky

```
 ==================== FAILURES ====================
                     _ FilesetRelPathWrapperTest.test_glob_mid_single _
                     
                     self = <pants_test.source.test_wrapped_globs.FilesetRelPathWrapperTest testMethod=test_glob_mid_single>
                     
                         def test_glob_mid_single(self):
                           self._spec_test('globs("a/*/Fleem.java")',
                     >                     {'globs': ['y/a/*/Fleem.java']})
                     
                     pants_test/source/test_wrapped_globs.py:76: 
                     _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                     pants_test/source/test_wrapped_globs.py:61: in _spec_test
                         graph = self.context().scan()
                     pants/goal/context.py:358: in scan
                         for address in self.address_mapper.scan_addresses(root):
                     pants/engine/legacy/address_mapper.py:100: in scan_addresses
                         return self._internal_scan_specs([DescendantAddresses(base_path)], missing_is_fatal=False)
                     _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                     
                     self = <pants.engine.legacy.address_mapper.LegacyAddressMapper object at 0x7f1f66e7ed68>
                     specs = [DescendantAddresses(directory='')]
                     fail_fast = True, missing_is_fatal = False
                     
                         def _internal_scan_specs(self, specs, fail_fast=True, missing_is_fatal=True):
                           # TODO: This should really use `product_request`, but on the other hand, we need to
                           # deprecate the entire `AddressMapper` interface anyway. See #4769.
                           request = self._scheduler.execution_request([BuildFileAddresses], [Specs(tuple(specs))])
                           returns, throws = self._scheduler.execute(request)
                         
                           if throws:
                             _, state = throws[0]
                             if isinstance(state.exc, (AddressLookupError, ResolveError)):
                               if missing_is_fatal:
                                 raise self.BuildFileScanError(
                                   'Spec `{}` does not match any targets.\n{}'.format(
                                     self._specs_string(specs), str(state.exc)))
                               else:
                                 # NB: ignore Throws containing ResolveErrors because they are due to missing targets / files
                                 return set()
                             else:
                     >         raise self.BuildFileScanError(str(state.exc))
                     E         pants.build_graph.address_mapper.AddressMapper.BuildFileScanError: PathGlobs expansion failed: While expanding link "z/w/y": Failed to read_link for Link("z/w/y"): Os { code: 2, kind: NotFound, message: "No such file or directory" }
                     
                     pants/engine/legacy/address_mapper.py:82: BuildFileScanError
```

Seen again on master release cut for 1.16.0.dev2 in the 'Unit tests (Py2.7 PEX)' shard: https://travis-ci.org/pantsbuild/pants/jobs/523154638#L733
```
                   Invalidated 1 target.
19:59:07 01:26       [run]
                     ........F......ssss.................       [100%]
                     ==================== FAILURES ====================
                     _ FilesetRelPathWrapperTest.test_glob_mid_single _
                     
                     self = <pants_test.source.test_wrapped_globs.FilesetRelPathWrapperTest testMethod=test_glob_mid_single>
                     
                         def test_glob_mid_single(self):
                           self._spec_test('globs("a/*/Fleem.java")',
                     >                     {'globs': ['y/a/*/Fleem.java']})
                     
                     pants_test/source/test_wrapped_globs.py:76: 
                     _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                     pants_test/source/test_wrapped_globs.py:61: in _spec_test
                         graph = self.context().scan()
                     pants/goal/context.py:370: in scan
                         for address in self.address_mapper.scan_addresses(root):
                     pants/engine/legacy/address_mapper.py:100: in scan_addresses
                         return self._internal_scan_specs([DescendantAddresses(base_path)], missing_is_fatal=False)
                     _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
                     
                     self = <pants.engine.legacy.address_mapper.LegacyAddressMapper object at 0x7f240fc17490>
                     specs = [DescendantAddresses(directory=u'')]
                     fail_fast = True, missing_is_fatal = False
                     
                         def _internal_scan_specs(self, specs, fail_fast=True, missing_is_fatal=True):
                           # TODO: This should really use `product_request`, but on the other hand, we need to
                           # deprecate the entire `AddressMapper` interface anyway. See #4769.
                           request = self._scheduler.execution_request([BuildFileAddresses], [Specs(tuple(specs))])
                           returns, throws = self._scheduler.execute(request)
                         
                           if throws:
                             _, state = throws[0]
                             if isinstance(state.exc, (AddressLookupError, ResolveError)):
                               if missing_is_fatal:
                                 raise self.BuildFileScanError(
                                   'Spec `{}` does not match any targets.\n{}'.format(
                                     self._specs_string(specs), str(state.exc)))
                               else:
                                 # NB: ignore Throws containing ResolveErrors because they are due to missing targets / files
                                 return set()
                             else:
                     >         raise self.BuildFileScanError(str(state.exc))
                     E         BuildFileScanError: PathGlobs expansion failed: While expanding link "z/w/y": Failed to read_link for Link("z/w/y"): Os { code: 2, kind: NotFound, message: "No such file or directory" }
                     
                     pants/engine/legacy/address_mapper.py:82: BuildFileScanError
                     - generated xml file: /home/travis/build/pantsbuild/pants/.pants.d/test/pytest/tests.python.pants_test.source.wrapped_globs/junitxml/TEST-tests.python.pants_test.source.wrapped_globs.xml -
                     ============ short test summary info =============
                     FAIL tests/python/pants_test/source/test_wrapped_globs.py::FilesetRelPathWrapperTest::test_glob_mid_single
                     SKIP [4] ../../../../../../usr/lib/python2.7/unittest/case.py:60: TODO: #4760
                     1 failed, 31 passed, 4 skipped in 1.57 seconds
```

I'm going to dig a bit and skip if nothing stands out.

--- a/tests/python/pants_test/source/test_wrapped_globs.py
+++ b/tests/python/pants_test/source/test_wrapped_globs.py
@@ -4,7 +4,6 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-import os
 import unittest
 from textwrap import dedent
 
@@ -46,7 +45,7 @@ class FilesetRelPathWrapperTest(TestBase):
     self.create_file('y/morx.java')
     self.create_file('y/fleem.java')
     self.create_file('z/w/foo.java')
-    os.symlink('../../y', os.path.join(self.build_root, 'z/w/y'))
+    self.create_link('y', 'z/w/y')
 
   def test_no_dir_glob(self):
     self.add_to_build_file('y/BUILD', 'dummy_target(name="y", sources=globs("*"))')

